src/licensedcode/match.py

```diff
@@ -1389,6 +1395,55 @@ def filter_already_matched_matches(matches, query):
     return kept, discarded
 
 
+def filter_key_phrase_spans(matches):
+    """
+    Return a filtered list of kept LicenseMatch matches and a list of
+    discardable matches by removing all matches that do not contain all key
+    phrases required by the rule.
```

Comment:
What if there is only one match and this does not contain the keyphrases?
Would you discard it, therefore leaving no matches?

Reply:
If the only match does not contain the key phrase wouldn't that indicate a high chance of being a false positive? If so, for my use-case I would rather have no results, but maybe that's not in-line with Scancode philosophy?

Reply:
> If the only match does not contain the key phrase wouldn't that indicate a high chance of being a false positive?

yes in the general case, but let's take a practical  example:

1. in query I have  `licensed under the AGPL 3.0` and in rule1 I have `AGPL 3.0`
2. in rule2 I have `licensed under the {{AGPL 3.0 license}}`
3. the query will initially match both rules
4. the match to rule1 may be discarded as contained in the rule2 match
5. the match to rule2 may be discarded as missing a key phrase

I am left with no matches yet `licensed under the AGPL 3.0` was very clearly a license notice.

On the other hand if the match to rule2 is discarded as missing a key phrase early and before the containment filter, then we are at least left with the match to rule1?

Reply:
Okay, that makes sense. But unfortunately I was unable to recreate this scenario in a unit test.

Instead I experimented and moved the key phrase filter up (before the containment filter) and 5 of the datadriven tests I've added broke. Looking into that now.

Also, are you suggesting to skip if there's only one rule like some of the other filters?

```
    if len(matches) < 2:
        return matches
```